src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+            self.polarization.subscribe(self._onPolarization)
+
+    def _project2Polar(self, pos):
+        """
+        Return the polar projection of the image at the given position.
+        :param pos: (float, float, string or None) position (must be part of the .stream._pos)
+        :returns: (DataArray) The polar projection
+        """
+        # Note: Need a copy of the link to the dict. If self._polarimetry_dict is reset while
+        # still running this method, the dict might get new entries again, though it should be empty.
+        polar_dict = self._polar_dict
+        # TODO BUg happens sometimes
+        # 15:40:23 ERROR   log            : Updating ARRawProjection image
+        # Traceback (most recent call last):
+        #   File "/home/sabrina/development/odemis/src/odemis/acq/stream/_projection.py", line 422, in _updateImage
+        #     polar_data = self._project2Polar(pos + (pol,))
+        #   File "/home/sabrina/development/odemis/src/odemis/acq/stream/_projection.py", line 356, in _project2Polar
+        #     polar_dict = self._polar_dict
+        # AttributeError: 'ARRawProjection' object has no attribute '_polar_dict'
+
+        if pos in polar_dict:
+            polar_data = polar_dict[pos]
+        else:
+            # Compute the polar representation
+            data = self.stream._pos[pos]
+            try:
+                # Correct image for background. It must match the polarization (defaulting to MD_POL_NONE).
+                data_bg_corr = self._processBackground(data, data.metadata.get(model.MD_POL_MODE, model.MD_POL_NONE))
+
+                if numpy.prod(data_bg_corr.shape) > (1280 * 1080):
+                    # AR conversion fails with very large images due to too much
+                    # memory consumed (> 2Gb). So, rescale + use a "degraded" type that
+                    # uses less memory. As the display size is small (compared
+                    # to the size of the input image, it shouldn't actually
+                    # affect much the output.
+                    logging.info("AR image is very large %s, will convert to "
+                                 "azimuthal projection in reduced precision.",
+                                 data_bg_corr.shape)
+                    y, x = data_bg_corr.shape
+                    if y > x:
+                        small_shape = 1024, int(round(1024 * x / y))
+                    else:
+                        small_shape = int(round(1024 * y / x)), 1024
+                    # resize raw data
+                    data_bg_corr = img.rescale_hq(data_bg_corr, small_shape)
+
+                # define the size of the image for polar representation in GUI
+                # 2 x size of original/raw image (on smallest axis) and at most
+                # the size of a full-screen canvas (1134)
+                size = min(min(data_bg_corr.shape) * 2, 1134)
+
+                # TODO still valid or remove due to improved runtime of triangulation code??
```

Comment:
I don't think that the code is a lot faster than the original code. However, it's definitely not something that's really useful to do because the user can typically wait a couple of seconds. So you can drop this TODO. 
The second TODO is also an interesting idea. It has a slightly higher chance to be one day implemented, but still very unlikely...